# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxHarfBuzzShaper.h
# Commit: cd5ac5f6ff6e
# Full Hash: cd5ac5f6ff6e626bd69e4d42ff9d1d7078abae03
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2022-03-01 09:40:29
# Regressor Bug: 1756720
# File Overlap Count: 1
# Description:
#   Bug 1756720 - When using legacy MS Symbol fonts, map character codes from the U+00xx range to the PUA range U+F0xx found in the cmap subtables. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D139724
# ==============================================================================

diff -r b547c1a2ef61 -r cd5ac5f6ff6e gfx/thebes/gfxHarfBuzzShaper.h
--- a/gfx/thebes/gfxHarfBuzzShaper.h	Mon Feb 28 20:34:33 2022 +0000
+++ b/gfx/thebes/gfxHarfBuzzShaper.h	Mon Feb 28 20:38:38 2022 +0000
@@ -169,6 +169,12 @@
   // Whether the font implements GetGlyph, or we should read tables
   // directly
   bool mUseFontGetGlyph;
+
+  // Whether the font is an MS Symbol-encoded font, in which case we will
+  // try remapping U+0020..00FF to U+F020..F0FF for characters in the U+00xx
+  // range that are otherwise unsupported.
+  bool mIsSymbolFont;
+
   // Whether the font implements GetGlyphWidth, or we should read tables
   // directly to get ideal widths
   bool mUseFontGlyphWidths;